### ISSUE TITLE
a question about same random seed but a little different result(train loss) on the same computer

### ISSUE THREAD

**Issue by clw5180** [open-mmlab/mmdetection#2034]
I try to use: 
python tools/train.py 'configs/dcn/cascade_rcnn_dconv_c3-c5_r50_fpn_1x.py' --validate **--seed=10 --deterministic**
I wish to get same results for different training process under same model config, in order to tune some hyparameters, and I find that the train loss is nearly the same but not  completely consistent. So anyone know what reason cause this difference?

**Comment by yhcao6** [comment]
The difference may be related to some ***atomic*** operation. You may refer to [Reproducibility](https://pytorch.org/docs/stable/notes/randomness.html).

**Comment by clw5180** [comment]
> The difference may be related to some _**atomic**_ operation. You may refer to [Reproducibility](https://pytorch.org/docs/stable/notes/randomness.html).

Thanks a lot! I already read something about it, and I think it's caused by the  backward process of F.interpolate. Is it very difficult to make it deterministic?

**Comment by yhcao6** [comment]
Sorry I don't know. 
I also found in the first iteration the loss is totally same, but at the second iteration the loss differs a little, the non-deterministic may come from the backward process.
